### PR TITLE
Created Email Script

### LINKED ISSUE #100: [Create] Weekly Report Script
## Task Description
- Create a script that creates a summary report of a given station as a pdf file and then emails it to email subscribers from the database.
## User Stories
- As an avid user of my local station, I'd be interested in automated weekly email reports to have a look at while I'm on the train or having my morning coffee.

### PR DESCRIPTION
**Pull request by isaacschaessenscoleman**
### Script for Emailed Reports
I've created a script which sends station-specific reports to email subscribers on execution (reports/report_emailer.py).

This script is split up into three sections:
**1. I've created a script which queries aggregate information on stations, producing summary statistics and graphs.** 
_Functions_:
- average_delay_per_hour_graph
- get_station_name
- get_delay_and_cancellation_percentages
- get_average_delay_time
- get_most_common_cancellation_reasons
- cancellation_types_pie_chart

**2. The script then uses this data to build a html script for a summary report for a given station. The html script is then converted into a pdf.**
_Functions_:
- generate_html
- convert_html_to_pdf

**3.  Finally, the pdf reports are emailed to the relevant subscribers.** 
_Functions_:
- generate_email_object
- email_sender
- group_email_subscribers

Still to do: Add a function which uploads the pdf reports to an s3 bucket after the emails have been sent.

Closes #100

### CODE REVIEW

File: reports/report_emailer.py [hunk 1]
```diff
@@ -63,22 +43,27 @@ def average_delay_per_hour_graph(conn: psycopg2.extensions.connection, station_c
                 y=[i[1] for i in delays_per_hour_data])
     plt.xlabel("Hour of the Day")
     plt.ylabel("Average Delay Time")
+    plt.title("Average Delay Time for Delayed Trains")
```
Comment (adamski201) [inline reports/report_emailer.py:46]: Would be good to specify a time period here. The reports are weekly, so it'd be good for the function to only use data from the last week.

File: reports/report_emailer.py [hunk 2]
```diff
@@ -63,22 +43,27 @@ def average_delay_per_hour_graph(conn: psycopg2.extensions.connection, station_c
...
 
     plt.savefig("average_delay_per_hour_graph.jpg")
 
 
-def get_station_name(conn: psycopg2.extensions.connection, station_crs: str):
+def get_station_name(conn: psycopg2.extensions.connection, station_crs: str) -> str:
```
Comment (adamski201) [inline reports/report_emailer.py:51]: Nitpick : Could use str | None as the return type here, or alternatively (or both) mention that None is a possible return type in the docstring.

File: reports/report_emailer.py [hunk 3]
```diff
@@ -158,8 +152,10 @@ def get_most_common_cancellation_reasons(conn: psycopg2.extensions.connection, s
     return cancellation_reasons
 
 
-def cancellation_types_pie_chart(conn: psycopg2.extensions.connection, station_crs: str):
-    """"""
+def cancellation_types_pie_chart(conn: psycopg2.extensions.connection, station_crs: str) -> None:
+    """This function creates a piec hart of the station's most common cancellation
```
Comment (adamski201) [inline reports/report_emailer.py:156]: Typo in "pie chart"

File: reports/report_emailer.py [hunk 4]
```diff
@@ -225,27 +224,133 @@ def generate_html(conn: psycopg2.extensions.connection, station_crs: str):
...
+    """This function sends an email via AWS SES. It accepts an email object (
+    i.e. what to email) and the addresses to email to."""
+
+    ses_client = client("ses",
+                        region_name="eu-west-2",
+                        aws_access_key_id=os.environ["AWS_ACCESS_KEY_ID"],
+                        aws_secret_access_key=os.environ["AWS_SECRET_ACCESS_KEY"])
+
+    ses_client.send_raw_email(
+        Source='trainee.isaac.schaessens.coleman@sigmalabs.co.uk',
```
Comment (adamski201) [inline reports/report_emailer.py:281]: nitpick: magic string could be placed at top of file

**Review by adamski201 — CHANGES_REQUESTED** [review]
Really good work, thanks for doing this. Just one function that needs to be fixed and aside from that just a couple of nitpicks and minor typos.

File: reports/report_emailer.py [hunk 5]
```diff
@@ -114,21 +99,25 @@ def get_delay_and_cancellation_percentages(conn: psycopg2.extensions.connection,
...
 
-    avg_delay_query = f"""SELECT 
+    avg_delay_query = f"""SELECT
                         ROUND(AVG((EXTRACT(EPOCH FROM (arrivals.actual_arrival - arrivals.scheduled_arrival)) / 60)), 2) AS average_delay_minutes 
                         FROM arrivals
                         INNER JOIN stations
                         ON stations.station_id=arrivals.station_id
                         WHERE stations.crs_code='{station_crs}'
+                        AND 
+                        arrivals.actual_arrival > arrivals.scheduled_arrival
```
Comment (aribasyeda) [inline reports/report_emailer.py:120]: If you prefer, you can store all of your SQL queries in separate Python file (such as sql_queries.py) as variables can indeed make your code more organised and easier to read

File: reports/report_emailer.py (hunk 2 above)
Comment (aribasyeda) [reply reports/report_emailer.py:51]: Good catch Adam

**Review by aribasyeda — COMMENTED** [review]
Great work Isaac. Nothing to much to add to Adam's comments

**Review by adamski201 — APPROVED** [review]

**Pull request merged**